modules/worker/pool_test.go

```diff
@@ -176,46 +181,36 @@ func TestPool_ContextCancellation(t *testing.T) {
 	unittest.RequireAllDone(t, pool)
 }
 
-// TestPool_JobPanic tests that if a job panics (throws irrecoverable error),
-// the pool captures the throw and continues operating.
+// TestPool_JobPanic tests that ThrowIrrecoverable properly panics when called.
 func TestPool_JobPanic(t *testing.T) {
```

Comment:
[nit]  `TestPool_JobPanic` no longer tests pool behavior with panics, only `ThrowIrrecoverable` directly. I'd suggest renaming to `TestThrowableContext_Panic`.